armi/utils/plotting.py

```diff
@@ -38,3 +53,724 @@ def colorGenerator(skippedColors=10):
+                facecolor=facecolor,
+            )
+            ax.add_artist(rectangle)
+            self.pageRects.append(rectangle)
+            ax.text(
+                float(i) / numpages + 0.5 / numpages,
+                0.5,
+                "{:.1f}".format(depth),
+                ha="center",
+                va="center",
+                transform=ax.transAxes,
+                fontsize=self.fontsize,
+            )
+        self.valtext.set_visible(False)
+
+        divider = axes_grid1.make_axes_locatable(ax)
+        bax = divider.append_axes("right", size="5%", pad=0.05)
+        fax = divider.append_axes("right", size="5%", pad=0.05)
+        self.button_back = matplotlib.widgets.Button(
+            bax, label="$\u25C0$", color=self.facecolor, hovercolor=self.activecolor
+        )
+        self.button_forward = matplotlib.widgets.Button(
+            fax, label="$\u25B6$", color=self.facecolor, hovercolor=self.activecolor
+        )
+        self.button_back.label.set_fontsize(self.fontsize)
+        self.button_forward.label.set_fontsize(self.fontsize)
+        self.button_back.on_clicked(self.backward)
+        self.button_forward.on_clicked(self.forward)
+
+    def _update(self, event):
+        super(PageSlider, self)._update(event)
+        i = int(self.val)
+        if i >= self.valmax:
+            return
+        self._colorize(i)
+
+    def _colorize(self, i):
+        for j in range(len(self.depths)):
+            self.pageRects[j].set_facecolor(self.facecolor)
+        self.pageRects[i].set_facecolor(self.activecolor)
+
+    def forward(self, event):
+        """Move forward to the next depth (page)."""
+        current_i = int(self.val)
+        i = current_i + 1
+        if (i < self.valmin) or (i >= self.valmax):
+            return
+        self.set_val(i)
+        self._colorize(i)
+
+    def backward(self, event):
+        """Move backward to the previous depth (page)."""
+        current_i = int(self.val)
+        i = current_i - 1
+        if (i < self.valmin) or (i >= self.valmax):
+            return
+        self.set_val(i)
+        self._colorize(i)
+
+
+def plotAssemblyTypes(
+    core, assems=None, plotNumber=1, maxAssems=None, showBlockAxMesh=True
+):
+    """
+    Generate a plot showing the axial block and enrichment distributions of each assembly type in the core.
+
+    Parameters
+    ----------
+    core: Core
+        The core to plot assembly types of.
+
+    assems: list
+        list of assembly objects to be plotted.
+
+    plotNumber: integer
+        number of uniquely identify the assembly plot from others and to prevent plots from being overwritten.
+
+    maxAssems: integer
+        maximum number of assemblies to plot in the assems list.
+
+    showBlockAxMesh: bool
+        if true, the axial mesh information will be displayed on the right side of the assembly plot.
+    """
+
+    if assems is None:
+        assems = list(core.parent.blueprints.assemblies.values())
```

Comment:
If the only thing that we are using the core for is to get the blueprints, then we should probably just pass the blueprints. Looking at the call site higher in the diff, I was confused to see that core was being passed at all, prompting me to dig in farther to see this.

Reply:
Good point. I just moved this out of reactors and didn't change anything, so I didn't spend much time reading it.